proto/reaction.proto

```diff
@@ -287,6 +288,14 @@ message ReactionRole {
     // A product can be any species produced by the reaction, whether desired
     // or undesired.
     PRODUCT = 8;
+    // When there is only one intended chemical equation.
```

Comment:
I'm having trouble following this comment. Maybe you mean something like:

```
// Use BYPRODUCT to indicate a chemical species that is an expected result of the reaction but is not the product of interest.
// Use SIDE_PRODUCT to indicate the product of a side reaction.
// See https://doi.org/10.1021/op300317g for a discussion of these terms.

Reply:
I feel the clause "When there is only one intended chemical equation" is important here: if there are more than one intended chemical equation and product A is a byproduct, I still don't know which equation A is a part of. One can use desired/side/byproduct roles to balance an equation only "when there is only one intended chemical equation" 

Also, is it possible (and should we allow it) for a product to be a byproduct and a side product at the same time?

Reply:
It's fine with me if you add that clause somewhere; is it common for there to be more than one? I guess I was assuming this was implied. I think that the definitions given in that article are mutually exclusive (assuming a single reaction is intended), so I think a single assignment is appropriate. Also, note that these reaction role assignments are not designed to be gospel truth; they are always going to be somewhat subjective.